e2e/nomostest/sync.go

```diff
@@ -263,3 +266,58 @@ func statusHasSyncDirAndNoErrors(status v1beta1.Status, sourceType configsync.So
 	}
 	return nil
 }
+
+func resourceGroupHasReconciled(sourceHash string) testpredicates.Predicate {
```

Comment:
Probably want some condition checks in here for Stalled at least. The Reconciling condition is now deprecated by https://github.com/GoogleContainerTools/kpt-config-sync/pull/1571 but it might be worth adding a quick inline comment about it here to remind us why we're not checking it.

Reply:
Actually, the StatusEquals Current check might handle the Stalled condition.

Reply:
Added an additional check to ensure it has the Stalled condition and it is False.